### PR TITLE
clean up

### PR DESCRIPTION
**Pull request by AndrewKirkovski**
* Migrated to yarn
* Separated types to other directory
* Cleaned up deprecated public folder
* Cleaned up unused/non working scripts

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ typings/*
 .env
 npm-debug.log
 .phpstorm.meta.php/
+tsconfig.tsbuildinfo
```
Comment (AndrewKirkovski) [inline .gitignore]: @Kreezag 
That file is used only for incremental builds and not needed in repo

File: package.json [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+    "type-check:watch": "yarn type-check -- --watch",
     "watch": "webpack --watch --colors",
     "build:js": "webpack",
-    "build:types": "tsc --emitDeclarationOnly",
-    "build": "npm run build:js && npm run build:types",
-    "start": "npm run watch && npm run type-check:watch",
-    "lint": "./node_modules/.bin/eslint ./**/*.ts --fix",
+    "build:types": "tsc --emitDeclarationOnly --outDir ./types",
+    "build": "yarn build:js && yarn build:types",
+    "lint": "eslint ./**/*.ts --fix",
```
Comment (AndrewKirkovski) [inline package.json]: @Kreezag using 'node_modules' is bad practice as you are not guaranteed it exists

File: types/lib/callbackRegistry.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+interface ICallback {
```
Comment (AndrewKirkovski) [inline types/lib/callbackRegistry.d.ts]: @Kreezag 
I moved all generated declarations in separate dir, this way they are not getting in a way during development
Keeping generated files separately from originals is a good idea generally

**Comment by wolfy-j** [comment]
Let me know when we can merge it.

**Pull request merged**